docs/jegyzet/ado_net/ADO_NET.md

```diff
@@ -0,0 +1,275 @@
+
+**Példa** kapcsolat létrehozására `StringBuilder`-rel:
+```cs
+var builder = new SqlConnectionStringBuilder();
+builder.UserID = "User";
+builder.Password = "Pw";
+builder.DataSource = "database.server.hu";
+builder.InitialCatalog = "Adatvez";
+
+var conn = new SqlConnection(builder.ConnectionString);
+conn.Open();
+... //lekérdezések, eljárások végrehajtása
+conn.Close(); //nem felejtjük el bezárni, hogy elkerüljük a Connection leak-et
+```
+
+## Command ##
+Az adatbázis kapcsolat létrehozása után szeretnénk az adatbázissal kommunikálni, lekérdezéseket, eljárásokat futtatni. Ehhez az **ADO** az `IDbCommand` interfészt biztosítja, ami egy utasítást reprezentál.
+
+### Előkészítés ###
+Az `IDbCommand` alábbi főbb *property*-jeit beállítva tudjuk megmondani, hogy a megadott parancs, hogyan legyen értelmezve, és miként viselkedjen az utasítás:
+- `CommandType`:  háromféle lehet
+-- Tárolt eljárás (*StoredProcedure*)
+-- Tábla teljes tartalmához hozzáférés (*TableDirect, nem minden implementáció használja, pl. OleDB igen*)
+-- SQL query (*Text*) - alapértelmezett
+- `CommandText`: maga a parancs szövege
+- `Connection`: az adatbázis kapcsolat
+- `Transaction`: a tranzakció
+- `CommandTimeout`: timeout arra az esetre, ha valami elakad (*alapértelmezetten 30s*)
+- `Parameters`: paraméterek, az [SQL injection támadás](#veszelyforrasok) kivédéséhez
+
+### Végrehajtás ###
+Miután előkészítettük a `Command`-ot, végre is hajthatjuk azt. Ehhez többféle lehetőség közül is választhatunk, attól függően, hogy mit várunk visszatérési értéknek.
+
+- `ExecuteReader`: több rekord lekérdezése 
+- `ExecuteScalar`: skalár érték lekérdezése
+- `ExecuteNonQuery`: olyan lekérdezés, amit nem eredményhalmazt ad vissza (pl. INSERT) - az érintett sorok számát adja helyette vissza
+- `ExecuteXmlReader` (MS SQL Server): XML dokumentumot (`XmlReader`* objektum*) ad vissza, az eredmény egy rekord egyetlen XML oszlopa 
+
+Parancsokat újrafelhasználni is tudunk, a `Command.Prepare()` (*szerver oldalon előkészíti a parancs futtatását*) hívás után, ám ez csak akkor éri meg, ha elegendően sokszor futtatjuk az utasítást, mivel az előkészítés és előkészülten tartás idő- és erőforrásigényes. 
+
+**Példa** parancs használatára:
+```cs
+//kapcsolatot létrehozása, megnyitása
+...
+var command = new SqlCommand();
+command.Connection = connection;
+command.CommandType = CommandType.StoredProcedure;
+command.CommandText = "salesByCategory"; //StoredProcedure, ezért elég a tárolt eljárás nevét megadni
+
+
+var parameter = new SqlParameter();
+parameter.ParameterName = "@CategoryName"; //ilyen néven kerül bele a lekérdezésbe
+parameter.SqlDbType = SqlDbType.NVarChar;
+parameter.Value = categoryName; 
+//valahol már létrehoztuk a categoryName változót
+
+command.Parameters.Add(parameter);
+//SQL injection elleni védelem
+
+var reader = command.ExecuteReader();

```

Comment:
olvasásra nem akarunk példát adni?

Reply:
ezt nem pontosan értem, az "Az eredmény" részben van példa adatok olvasására DataSet és DataReaderrel is

Reply:
ok jó lesz így